apps/frontend/src/libs/components/input/libs/components/toggle-vision-button/toggle-vision-button.tsx

```diff
@@ -0,0 +1,44 @@
+import { type MouseEvent, useCallback } from "react";
+
+import { Icon } from "~/libs/components/icon/icon.js";
+
+import styles from "./styles.module.css";
+
+type Properties = {
+	onAppearanceToggle: () => void;
+	type: "password" | "text";
+};
+
+const ToggleVisionButton = ({
+	onAppearanceToggle,
+	type,
+}: Properties): React.JSX.Element => {
+	const typeStateToPasswordIcon: "eye" | "eyeOff" =
```

Comment:
typeStateToPasswordIcon sounds more like a map for me 🤔 

```suggestion
	const iconName: "eye" | "eyeOff" =
```